data/preprocess/preprocess_wn11.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python
+"""Preprocess the WN11 dataset into a the format expected by libKGE. """
+
+import argparse
+import yaml
+import os.path
+import numpy as np
+from collections import OrderedDict
+
+from util import analyze_raw_splits
+from util import RawDataset
+from util import write_dataset_config
+from util import RawSplit
+from util import PosNegRawSplit
+from util import process_splits
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("folder", type=str)
+    args = parser.parse_args()
+
+    print(f"Preprocessing {args.folder}...")
+
+    raw_train = RawSplit(
+        file="train.txt",
+        collect_entities=True,
+        collect_relations=True,
+        derived_split_key="train",
+        derived_split_options={
+            "type": "triples",
+            "filename": "train.del",
+            "split_type": "train",
+        },
+        derived_sample_split_key="train_sample",
+        sample_split_options={
+            "type": "triples",
+            "filename": "train_sample.del",
+            "split_type": "train",
+        },
```

Comment:
Make this a list of DerivedSplits. In this way users can easily extend the preprocessing with a custom derived split without any hacks.